### PR TITLE
feat(CC): add a datasource to get the list of CC network instances

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add data source network instances
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add data source network instances
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/cc" TESTARGS="-run TestAccDataSourceCcNetworkInstances_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cc -v -run TestAccDataSourceCcNetworkInstances_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceCcNetworkInstances_basic
=== PAUSE TestAccDataSourceCcNetworkInstances_basic
=== CONT  TestAccDataSourceCcNetworkInstances_basic
--- PASS: TestAccDataSourceCcNetworkInstances_basic (136.83s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cc        136.878s

```

### CODE REVIEW

File: docs/data-sources/cc_network_instances.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+* `name` - (Optional, String) Specifies the network instance name.
+
+* `description` - (Optional, String) Specifies the network instance description.
+
+* `cloud_connection_id` - (Optional, String) Specifies the cloud connection ID.
+
+* `status` - (Optional, String) Specifies the status of the network instance.
+  The options are as follows:
+  + **ACTIVE**: The network instance is available.
```
Comment (houpeng80) [inline docs/data-sources/cc_network_instances.md:36]: PENDING, ERROR are supported

File: huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_network_instances_test.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	return fmt.Sprintf(`
+resource "huaweicloud_vpc" "test1" {
+  name = "%[1]s1"
+  cidr = cidrsubnet("10.12.0.0/16", 4, 1)
+}
+
+resource "huaweicloud_vpc" "test2" {
+  name = "%[1]s2"
+  cidr = cidrsubnet("192.168.0.0/16", 4, 1)
+}
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_network_instances_test.go]:
```suggestion
resource "huaweicloud_vpc" "test1" {
  count = 2
  name = "%[1]s_${count.index}"
  cidr = cidrsubnet("10.12.0.0/16", 4, 1)
}
```
fix huaweicloud_vpc_subnet, huaweicloud_cc_network_instance like this

File: docs/data-sources/cc_network_instances.md (hunk 1 above)
Comment (profoundwu) [reply docs/data-sources/cc_network_instances.md:36]: fixed

File: huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_network_instances_test.go (hunk 2 above)
Comment (profoundwu) [reply huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_network_instances_test.go]: fixed

**Pull request merged**